api/oc_mnt.c

```diff
@@ -62,11 +62,21 @@ post_mnt(oc_request_t *request, oc_interface_mask_t iface_mask, void *data)
   }
 
   if (success) {
+#if defined(OC_DYNAMIC_ALLOCATION) && defined(OC_REP_ENCODING_REALLOC)
+    oc_rep_new_realloc(&request->response->response_buffer->buffer,
+                       request->response->response_buffer->buffer_size,
+                       OC_MAX_APP_DATA_SIZE);
+#else  /* OC_DYNAMIC_ALLOCATION && OC_REP_ENCODING_REALLOC */
     oc_rep_new(request->response->response_buffer->buffer,
-               OC_MAX_APP_DATA_SIZE);
+               request->response->response_buffer->buffer_size);
+#endif /* !OC_DYNAMIC_ALLOCATION || !OC_REP_ENCODING_REALLOC */
     oc_rep_start_root_object();
     oc_rep_set_boolean(root, fr, false);
     oc_rep_end_root_object();
+#ifdef OC_DYNAMIC_ALLOCATION
```

Comment:
Shouldn't this have the same `ifdef` as above?

```defined(OC_DYNAMIC_ALLOCATION) && defined(OC_REP_ENCODING_REALLOC)```

Reply:
Actually, doesn't need either `oc_rep_shrink_encoder_buf` checks that both are defined and does nothing if they aren't.

Reply:
We want to use shrink buffer for DYNAMIC memory so I changed the implementation.
OC_REP_ENCODING_REALLOC - just allows you to change OC_MIN_APP_DATA_SIZE:

- if OC_REP_ENCODING_REALLOC is set, the default value of OC_MIN_APP_DATA_SIZE is 256
- if OC_REP_ENCODING_REALLOC  is not set, the default value of OC_MIN_APP_DATA_SIZE is OC_MAX_APP_DATA_SIZE

Reply:
Sounds good to me.